### PR TITLE
LPS-204467: Allow some Object Entry fields that must not be forbidden

### PR DESCRIPTION
**Pull request by sergiojimcos**
**What is new?**
- Update some constants to use the Object Definition API and manage Object Entry REST Application name
- Update FieldTable to update those fields and switch to bussinessType

**JIRA Ticket**
https://liferay.atlassian.net/browse/LPS-204467

### CODE REVIEW

**Comment by sergiojimcos** [comment]
ci:test:relevant

**Comment by sergiojimcos** [comment]
ci:test:sf

**Comment by sergiojimcos** [comment]
ci:test:sf

**Comment by sergiojimcos** [comment]
ci:test:sf

**Comment by sergiojimcos** [comment]
ci:test:relevant

File: modules/apps/batch-planner/batch-planner-web/src/main/resources/META-INF/resources/js/constants.js [hunk 1]
```diff
@@ -60,6 +61,10 @@ export const IMPORT_STRATEGY_SELECTED_EVENT = 'ie-import-strategy-selected';
 export const JSON_FORMAT = 'json';
 export const JSONL_FORMAT = 'jsonl';
 export const NULL_TEMPLATE_VALUE = '';
+export const OBJECT_ENTRY_ENTITY_NAME_PREFIX =
```
Comment (4lejandrito) [inline modules/apps/batch-planner/batch-planner-web/src/main/resources/META-INF/resources/js/constants.js:64]: Hey @sergiojimcos can we use a similar approach to what @jaimelr10 did for the forbidden fields? I think we should not use anything objects specific in the code, instead have it all centralized.

**Review by 4lejandrito — CHANGES_REQUESTED** [review]

File: modules/apps/batch-planner/batch-planner-web/src/main/resources/META-INF/resources/js/constants.js (hunk 1 above)
Comment (4lejandrito) [reply modules/apps/batch-planner/batch-planner-web/src/main/resources/META-INF/resources/js/constants.js:64]: Let's fix the root issue: https://github.com/liferay-headless/liferay-portal/pull/1779#issuecomment-1869467606

**Comment by 4lejandrito** [comment]
You can use this curls to create universities with both field types:

```bash
curl 'http://localhost:8080/o/object-admin/v1.0/object-definitions/by-external-reference-code/University' -X PUT -H "authorization: Basic dGVzdEBsaWZlcmF5LmNvbTp0ZXN0" -H "content-type: application/json" --data-binary '{"active":true,"status":{"code":0},"objectFields":[{"DBType":"String","indexed":true,"indexedLanguageId":"","externalReferenceCode":"university-name","listTypeDefinitionId":0,"label":{"en_US":"University name"},"type":"String","required":true,"system":false,"name":"universityName","indexedAsKeyword":false},{"DBType":"BigDecimal","indexed":true,"indexedLanguageId":"","businessType":"PrecisionDecimal","externalReferenceCode":"university-year","listTypeDefinitionId":0,"label":{"en_US":"Year"},"name":"year","type":"BigDecimal","required":true,"system":false,"indexedAsKeyword":false},{"DBType":"String","businessType":"Picklist","defaultValue":"public","externalReferenceCode":"university-type","indexed":true,"indexedAsKeyword":false,"label":{"en_US":"University type"},"listTypeDefinitionExternalReferenceCode":"university-type-picklist","name":"universityType","objectFieldSettings":[{"name":"stateFlow","value":{"objectStates":[{"key":"public","objectStateTransitions":[{"key":"private"}]},{"key":"private","objectStateTransitions":[{"key":"public"}]}]}}],"required":true,"state":true,"system":false,"type":"String"}],"pluralLabel":{"en-US":"Universities"},"portlet":true,"scope":"company","panelCategoryKey":"control_panel.users","label":{"en-US":"University"},"externalReferenceCode":"University","name":"University"}'
curl 'http://localhost:8080/o/c/universities/by-external-reference-code/Cambridge' -X PUT -H "authorization: Basic dGVzdEBsaWZlcmF5LmNvbTp0ZXN0" -H "content-type: application/json" --data-binary '{"universityName":"Cambridge","year":1989,"universityType":"private"}'
curl 'http://localhost:8080/o/c/universities/by-external-reference-code/Oxford' -X PUT -H "authorization: Basic dGVzdEBsaWZlcmF5LmNvbTp0ZXN0" -H "content-type: application/json" --data-binary '{"universityName":"Oxford","year":1989,"universityType":"private"}'
```

**Comment by sergiojimcos** [comment]
ci:test:sf

File: modules/apps/batch-planner/batch-planner-web/src/main/resources/META-INF/resources/js/constants.js [hunk 2]
```diff
@@ -33,20 +33,18 @@ export const FILE_EXTENSION_INPUT_PARTIAL_NAME = 'externalType';
 export const FILE_SCHEMA_EVENT = 'file-schema';
 export const FORBIDDEN_CSV_FIELDS_ENTITY_NAMES = {
 	'com.liferay.object.rest.dto.v1_0.ObjectEntry': [
+		'actions',
```
Comment (4lejandrito) [inline modules/apps/batch-planner/batch-planner-web/src/main/resources/META-INF/resources/js/constants.js:36]: Move this logic to the openapi contributor in the backend.

File: modules/apps/object/object-rest-impl/src/main/java/com/liferay/object/rest/internal/openapi/v1_0/ObjectEntryOpenAPIResourceImpl.java [hunk 3]
```diff
@@ -148,7 +150,11 @@ private DTOProperty _getDTOProperty(ObjectField objectField) {
 				ObjectFieldConstants.BUSINESS_TYPE_ATTACHMENT)) {
 
 			DTOProperty dtoProperty = new DTOProperty(
-				Collections.singletonMap("x-parent-map", "properties"),
+				HashMapBuilder.<String, Object>put(
+					"x-batch-support", "false"
```
Comment (4lejandrito) [inline modules/apps/object/object-rest-impl/src/main/java/com/liferay/object/rest/internal/openapi/v1_0/ObjectEntryOpenAPIResourceImpl.java:154]: Rename this to `x-batch-csv-enabled`.

**Review by 4lejandrito — CHANGES_REQUESTED** [review]

**Comment by 4lejandrito** [comment]
ci:test:relevant

**Comment by 4lejandrito** [comment]
ci:test:sf

**Comment by 4lejandrito** [comment]
ci:forward

**Review by 4lejandrito — APPROVED** [review]
See my changes. I reviewed it with @ccorreagg and we decided this way was better.

**Comment by jalber786** [comment]
ci:test:relevant

**Comment by sergiojimcos** [comment]
ci:test:relevant

**Comment by sergiojimcos** [comment]
ci;forward

**Comment by sergiojimcos** [comment]
ci:forward

**Comment by yanan** [comment]
here are the [compare run results](https://testray.liferay.com/home/-/testray/runs/compare?compare=true&view=details&statusA=3&testrayRunIdB=2481093077&testrayRunIdA=2481389610&statusB=2), and the following seem related
- [rest-builder-jdk8](https://testray.liferay.com/home/-/testray/case_results/2481391135)
- [LocalFile.ImportAndExportObjects#ExportCustomObjectWithAllFieldTypesMapped](https://testray.liferay.com/home/-/testray/case_results/2481390291)

**Comment by sergiojimcos** [comment]
ci:forward

**Comment by sergiojimcos** [comment]
ci:forward

**Comment by yanan** [comment]
There are two failures seem related
- [rest-builder-jdk8](https://testray.liferay.com/home/-/testray/case_results/2482232190)
    - when I run buildREST, the `Field.java` is updated with `@JsonIgnore` added. This failed in both relevant and stable test suites  
image

- [LocalFile.ImportAndExportObjects#ExportCustomObjectWithAllFieldTypesMapped](https://testray.liferay.com/home/-/testray/case_results/2482231460)
    - the picklist field is not present in export C_Object with JSON wizard, on master we can select the picklist type field to export in JSON(picklist is not supported when export in CSV on master)

**Comment by sergiojimcos** [comment]
ci:forward

**Comment by sergiojimcos** [comment]
Sent to Brian here: https://github.com/brianchandotcom/liferay-portal/pull/145970

**Pull request closed without merging**